rsp_vision/save/save_data.py

```diff
@@ -0,0 +1,378 @@
+            analysis_log = pd.read_csv(f, index_col=0, header=0)
+            if analysis_log.empty:
+                # CASE 1: no data at all
+                sub = 0
+                ses = 0
+            elif analysis_log[
+                (analysis_log["mouse line"] == folder_naming_specs.mouse_line)
+                & (analysis_log["mouse id"] == folder_naming_specs.mouse_id)
+            ].empty:
+                # CASE 2: this subject was never analysed before
+                sub = analysis_log["sub"].max() + 1
+                ses = 0
+            elif analysis_log[
+                analysis_log["folder name"] == folder_naming_specs.folder_name
+            ].empty:
+                #  CASE 3: this subject was analysed before, but not this
+                #  session folder_name is the specific dataset name,
+                #  containing both info about the specific mouse and the
+                #  specific session this is why we use it to determine if
+                #  this session was analysed before
+                this_line_rows = analysis_log[
+                    (
+                        analysis_log["mouse line"]
+                        == folder_naming_specs.mouse_line
+                    )
+                    & (
+                        analysis_log["mouse id"]
+                        == folder_naming_specs.mouse_id
+                    )
+                ]
+                assert len(this_line_rows) >= 1
+                assert len(this_line_rows["sub"].unique()) == 1
+                sub = this_line_rows["sub"].unique()[0]
+
+                ses = (
+                    analysis_log[
+                        (
+                            analysis_log["mouse line"]
+                            == folder_naming_specs.mouse_line
+                        )
+                        & (
+                            analysis_log["mouse id"]
+                            == folder_naming_specs.mouse_id
+                        )
+                    ]["ses"].max()
+                    + 1
+                )
+            else:
+                # CASE 4: this session was analysed before
+                this_line_rows = analysis_log[
+                    (
+                        analysis_log["mouse line"]
+                        == folder_naming_specs.mouse_line
+                    )
+                    & (
+                        analysis_log["mouse id"]
+                        == folder_naming_specs.mouse_id
+                    )
+                ]
+                assert len(this_line_rows) >= 1
+                assert len(this_line_rows["sub"].unique()) == 1
+                sub = this_line_rows["sub"].unique()[0]
+
+                ses = this_line_rows["ses"].max()
+                reanalysis = True
+
+    except FileNotFoundError:
+        #  CASE 0: no analysis log file
+        sub = 0
+        ses = 0
+        analysis_log = pd.DataFrame(
+            columns=[
```

Comment:
I would not use spaces in column names, to be on the safe side. I would replace spaces with underscores (snake_case), as recommended [here](https://swc-blueprint.neuroinformatics.dev/specification.html#metadata-conventions)